_pylibczi/Reader.cpp

```diff
@@ -238,13 +232,28 @@ namespace pylibczi {
               c = vi[i].first;
               condensed.insert(vi[i].second);
           }
-          charSizes.emplace_back(c, condensed.size());
+          // added special case for images with multiple channels present
+          if (c=='C' && condensed.size()==1 && image_shape.size()==3)
+              charSizes.emplace_back(c, image_shape.end()[-3]); // channels
+          else
+              charSizes.emplace_back(c, condensed.size());
```

Comment:
>I'm assuming you're attempting to deal with BRG24 / BGR48 / BGR 96 pixel types?
I handle this with the flatten option in the Image class. That may not do what you need but I'm not entirely sure.

Correct.

The flatten flag did not work for the sample BGR file that I have. At a minimum, the arguments ``m_index`` and ``flatten`` to ``self.reader.read_selected(`` in CziFile.py are inverted. After I fixed this, I still was not able to read my sample non-mosaic BGR sample. I have shared this one via a dropbox link with you; it's about 30M in size, could be cropped further if you like. The existing release, because the channels are stored interleaved in the data, and not as true separate channels did not work. Maybe can you verify that you can get this file to work with your release, and we can use a cleaner fix? If it can work this way, and all we need to do is expose the flatten flag, I'm also fine with that.

Reply:
Sorry about the swapped flags I fixed that in one of the other branches, it's currently being reviewed. I'll see what I can do with the demo file. Thanks for sharing the file.